platforms/core-runtime/launcher/src/main/java/org/gradle/launcher/cli/BuildActionsFactory.java

```diff
@@ -77,42 +98,64 @@ public BuildActionsFactory(ServiceRegistry loggingServices) {
             .build();
         this.loggingServices = loggingServices;
         fileCollectionFactory = basicServices.get(FileCollectionFactory.class);
-        parametersConverter = new ParametersConverter(new BuildLayoutFactory(), basicServices.get(FileCollectionFactory.class));
+        buildEnvironmentConfigurationConverter = new BuildEnvironmentConfigurationConverter(
+            new BuildLayoutFactory(),
+            fileCollectionFactory,
+            basicServices.get(PropertyFactory.class)
+        );
         jvmVersionDetector = basicServices.get(JvmVersionDetector.class);
+        daemonJavaToolchainQueryService = new DaemonJavaToolchainQueryService(
```

Comment:
All of the `basicServices.get(Foo.class)` calls here make me think most of this code should itself live as a service in `BasicGlobalScopeServices`.

Reply:
Unfortunately those cannot be declared on `BasicGlobalScopeServices` since `core` doesn't has access to `JdkCacheDirectory`. However, I created `DaemonConfigurationServices` linked to configuration package in order to declare those ones. This seems something followed in the Gradle code where for example [DaemonClientServices](https://github.com/gradle/gradle/blob/f8ee6cae5bd056483bafec72043744540f532e09/platforms/core-runtime/launcher/src/main/java/org/gradle/launcher/daemon/client/DaemonClientServices.java) is linked to the `daemon.client` package